redfish-core/lib/power_subsystem.hpp

```diff
@@ -41,35 +61,25 @@ inline void getPowerSubsystemAllocation(
                 return;
             }
 
-            uint32_t powerCap{0};
-            bool powerCapEnable{false};
-            uint32_t maxPowerCapValue{0};
+            const uint32_t* powerCap = nullptr;
+            const bool* powerCapEnable = nullptr;
             const bool success = sdbusplus::unpackPropertiesNoThrow(
                 dbus_utils::UnpackErrorPrinter(), propertiesList, "PowerCap",
-                powerCap, "PowerCapEnable", powerCapEnable, "MaxPowerCapValue",
-                maxPowerCapValue);
+                powerCap, "PowerCapEnable", powerCapEnable);
 
             if (!success)
             {
                 messages::internalError(asyncResp->res);
                 return;
             }
 
-            // If MaxPowerCapValue valid, store Allocation properties in JSON
-            if ((maxPowerCapValue > 0) && (maxPowerCapValue < UINT32_MAX))
+            if ((powerCap != nullptr) && (powerCapEnable != nullptr))
             {
-                if (powerCapEnable)
-                {
-                    asyncResp->res.jsonValue["Allocation"]["AllocatedWatts"] =
-                        powerCap;
-                }
-                else
+                if (*powerCapEnable)
                 {
                     asyncResp->res.jsonValue["Allocation"]["AllocatedWatts"] =
-                        maxPowerCapValue;
+                        *powerCap;
```

Comment:
On further reflection I realize this won't be returning a consistent value for "AllocatedWatts". Both of the places which are filling in the value are asynchronous handlers. So there is no guarantee on the order that the value will be filled in.

Before I was thinking this was just an issue for the error path. But I actually believe it is an issue for the normal path as well. I do think these calls are going to have to be coordinated to assure a consistent value is returned.